### PR TITLE
ABIEOS_NO_INT128 option

### PR DESCRIPTION
**Pull request by tbfleming**
ABIEOS_NO_INT128 disables use of `__int128` for platforms which don't support it.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +37,10 @@ find_package(Threads)
 add_library(abieos STATIC src/abi.cpp src/crypto.cpp include/eosio/fpconv.c)
 target_include_directories(abieos PUBLIC include external/rapidjson/include)
 
+if(ABIEOS_NO_INT128)
+    target_compile_options(abieos PUBLIC -DABIEOS_NO_INT128)
```
Comment (revl) [inline CMakeLists.txt:41]: target_compile_definitions(abieos PUBLIC ABIEOS_NO_INT128)?

File: CMakeLists.txt [hunk 2]
```diff
@@ -47,9 +52,12 @@ add_executable(test_abieos src/test.cpp src/abieos.cpp)
 target_link_libraries(test_abieos abieos ${CMAKE_THREAD_LIBS_INIT})
 add_test(NAME test_abieos COMMAND test_abieos)
 
-add_executable(test_abieos_template src/template_test.cpp src/abieos.cpp)
-target_link_libraries(test_abieos_template abieos ${CMAKE_THREAD_LIBS_INIT})
-add_test(NAME test_abieos_template COMMAND test_abieos_template)
+if(ABIEOS_NO_INT128)
```
Comment (revl) [inline CMakeLists.txt:55]: if(NOT ABIEOS_NO_INT128)?

Comment (tbfleming) [reply CMakeLists.txt:55]: I wish. cmake bombed with that.

File: CMakeLists.txt (hunk 1 above)
Comment (tbfleming) [reply CMakeLists.txt:41]: changed

File: CMakeLists.txt (hunk 2 above)
Comment (tbfleming) [reply CMakeLists.txt]: It didn't work before but now it does. weird. changed.

**Pull request merged**